plugin/src/gatsby-node.ts

```diff
@@ -149,23 +151,30 @@ export async function sourceNodes(
   gatsbyApi: SourceNodesArgs,
   pluginOptions: ShopifyPluginOptions
 ) {
-  const lastOperationId = await gatsbyApi.cache.get(
-    LAST_SHOPIFY_BULK_OPERATION
-  );
+  const cacheKey = LAST_SHOPIFY_BULK_OPERATION + pluginOptions.typeName;
+  const lastOperationId = await gatsbyApi.cache.get(cacheKey);
 
   if (lastOperationId) {
     gatsbyApi.reporter.info(`Cancelling last operation: ${lastOperationId}`);
     await createOperations(pluginOptions, gatsbyApi).cancelOperation(
       lastOperationId
     );
-    await gatsbyApi.cache.set(LAST_SHOPIFY_BULK_OPERATION, undefined);
+    await gatsbyApi.cache.set(cacheKey, undefined);
   }
 
   const lastBuildTime = gatsbyApi.store.getState().status.plugins?.[
     `gatsby-source-shopify-experimental`
   ]?.lastBuildTime;
 
-  if (lastBuildTime) {
+  const hasBuiltFor = gatsbyApi.store.getState().status.plugins?.[
+    `gatsby-source-shopify-experimental`
+  ]?.hasBuiltFor; 
+
+  if (
+    lastBuildTime !== undefined && 
+    hasBuiltFor !== undefined && 
+    hasBuiltFor.includes(pluginOptions.typeName)
+  ) {
```

Comment:
I think `sourceChangedNodes` uses a cache key to retrieve `LAST_BUILD_TIME`. I think this value will be overwritten by the different instances of the plugin.

Reply:
Fixed! ✨